text/0039-sdk-performance-issue-creation.md

```diff
@@ -0,0 +1,72 @@
+* Start Date: 2022-11-28
+* RFC Type: decision
+* RFC PR: [#33](https://github.com/getsentry/rfcs/pull/33)
+* RFC Status: draft
+* RFC Driver: [Philipp Hofmann](https://github.com/philipphofmann)
+
+# Summary
+
+This RFC aims to clarify which types of performance issues SDKs shall create.
+
+# Motivation
+
+On June 21, 2022, we decided with [DACI](https://www.notion.so/sentry/Performance-Issue-Creation-POC-e521772ebccb482b83b08f4f8a3db2cb) to create performance issues in Ingest. While implementing the file I/O on the main thread performance issue for Android, the question arose as to why SDKs don't report easy-to-detect and matured performance issues. 
+
+# Options Considered
+
+## Option 1: SDKs report easy to detect and matured performance issues
+
+We still need to clarify what easy to detect and matured precisely means, which is not the goal of this document. For example, file I/O on the main thread could be a candidate.
+
+To clarify the threshold and configuration for performance issues, an experimental feature phase can help to get feedback.
+
+### Pros
+
+1. SDKs can capture a stack trace which will help with actionability.
+2. No running transaction required.
+3. Sentry can tie together transaction and error far more easily since both objects exist at the point of time the performance issue will be created.
+4. As we limit this only to easy detect peformance issues the SDK overhead should be minimal.
+
+### Cons
+
+1. Need for per-SDK rollout.
+2. Double dipping quotas, sending the transaction and the error created within that transaction.
```

Comment:
This might depend on how we create the performance issue, as IMO it shouldn't be an error anyway.

Reply:
I added an extra option 2 and added this as a pro there.